core/src/main/scala/org/scalacheck/ops/GenOrThrow.scala

```diff
@@ -13,15 +13,14 @@ import scala.util.Try
  * @param gen a generator of some kind
  * @tparam T the type of value produced by the generator
  */
+@deprecated("Use GenFromConfig instead. This class requires a ClassTag and is not as reusable as GenFromConfig. " +
+  "It will be removed in the next major version.", "2.7.0")
 class GenOrThrow[T: ClassTag](val gen: Gen[T]) {
```

Comment:
This old set of extension methods is no longer used, but I can't remove it because I need to retain binary compatibility.

Reply:
Can we mark things that are being retained for binary compatibility?  I don't know whether everyone thinks about these things and having a note in the code as to why gives anyone following behind either a reminder or a junior an opportunity to learn something from the codebase.